server/src/main/java/org/opensearch/cluster/routing/OperationRouting.java

```diff
@@ -300,7 +338,18 @@ private ShardIterator shardRoutings(
         @Nullable ResponseCollectorService collectorService,
         @Nullable Map<String, Long> nodeCounts
     ) {
-        if (ignoreAwarenessAttributes()) {
+        WeightedRoutingMetadata weightedRoutingMetadata = null;
+        if (clusterService != null) {
```

Comment:
is this more of a defensive check?

Reply:
Yes. This helps fixing older unit tests where otherwise we need to set ClusterService.